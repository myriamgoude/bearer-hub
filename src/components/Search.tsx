import * as React from 'react'
import Link from './Link'
import { path } from '../services/Integration'

import {
  ClearRefinements,
  Configure,
  Highlight,
  Hits,
  InstantSearch,
  Pagination,
  RefinementList,
  SearchBox
} from 'react-instantsearch-dom'

export default class Search extends React.Component {
  hit({ hit }: any) {
    return (
      <Link to={`${path({ id: hit.id, title: hit.title })}`}>
        <div>
          <h4>
            <Highlight attribute="title" hit={hit} />
          </h4>
        </div>
      </Link>
    )
  }

  render() {
    return (
      <div>
        <InstantSearch
          appId={`${process.env.GATSBY_ALGOLIA_APP_ID}`}
          apiKey={`${process.env.GATSBY_ALGOLIA_SEARCH_API_KEY}`}
          indexName={`${process.env.GATSBY_ALGOLIA_INDEX_NAME}`}
        >
          <div>
            <ClearRefinements />
            <h2>Category</h2>
            <RefinementList attribute="categories.title" />
            <h2>Provider</h2>
            <RefinementList attribute="providers.title" />
            <Configure hitsPerPage={8} />
          </div>
          <div>
            <SearchBox />
            <Hits hitComponent={this.hit} />
            <Pagination />
          </div>
        </InstantSearch>
      </div>
    )
  }
}
