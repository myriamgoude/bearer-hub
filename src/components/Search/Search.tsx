import * as React from 'react'

import { Configure, connectMenu } from 'react-instantsearch-dom'

import { CustomHits } from './components/CustomHits'
import { CustomSearchBox } from './components/CustomSearchBox'

interface ISearchProps {
  defaultCategory?: string
  defaultProvider?: string
}

export default class Search extends React.Component<ISearchProps> {
  render() {
    const categoriesAttribute = 'categories.title'
    const providersAttribute = 'providers.title'

    const VirtualMenu = connectMenu(() => null)

    return (
      <div>
        <div>
          <CustomSearchBox />
          <CustomHits hit={null} attribute={''} />

          {this.props.defaultCategory ? (
            <VirtualMenu attribute={categoriesAttribute} defaultRefinement={this.props.defaultCategory} />
          ) : null}

          {this.props.defaultProvider ? (
            <VirtualMenu attribute={providersAttribute} defaultRefinement={this.props.defaultProvider} />
          ) : null}
          <Configure hitsPerPage={8} />
        </div>
      </div>
    )
  }
}
