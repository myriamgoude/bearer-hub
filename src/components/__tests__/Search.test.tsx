import * as React from 'react'
import renderer from 'react-test-renderer'
import Search from '../Search'
import { InstantSearch } from 'react-instantsearch-dom'

const searchClient = {
  async search(requests: any) {
    return {
      results: requests.map(() => {
        // faking the return so we can render
        return {
          processingTimeMS: 0,
          nbHits: 0,
          hits: [],
          facets: {}
        }
      })
    }
  },
  async searchForFacetValues(requests: any) {
    return {
      results: requests.map(() => {
        return {}
      })
    }
  }
}

describe('Search', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        // @ts-ignore typings say apiKey & appId are required, but they aren't
        <InstantSearch indexName="test" searchClient={searchClient}>
          <Search />
        </InstantSearch>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
