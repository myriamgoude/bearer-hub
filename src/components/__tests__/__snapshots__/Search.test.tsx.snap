// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="ais-InstantSearch__root"
>
  <div>
    <div>
      <form
        action=""
        css={
          Object {
            "map": undefined,
            "name": "1dvxqr0",
            "next": undefined,
            "styles": "
  height: 72px;
  width: 512px;
  border: 1px solid #d4d9ea;
  border-radius: 4px;
  background-color: #ffffff;
  box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.49);
  position: absolute;
  top: -250px;
  left: 0;
  right: 0;
  margin: auto;

  input {
    appearance: none;
    width: 100%;
    height: 100%;
    background: transparent url(test-file-stub) no-repeat center left 1em / 32px;
    border: none;
    padding: 1em 1em 1em 3.5em;

    &:focus {
      outline-offset: 2px;
    }
  }
",
          }
        }
        noValidate={true}
        role="search"
      >
        <input
          onChange={[Function]}
          placeholder="Search an Integration"
          type="search"
          value=""
        />
      </form>
      <div
        css={
          Array [
            Object {
              "map": undefined,
              "name": "1n3w5wq",
              "next": undefined,
              "styles": "
    display: flex;
    margin: auto;
    flex-direction: column;
    justify-content: space-between;
    flex-wrap: wrap;

    @media (min-width: 576px) {
      flex-direction: row;
    }
  ",
            },
            false,
            false,
            undefined,
            Object {
              "map": undefined,
              "name": "1d7td7u",
              "next": undefined,
              "styles": "
          justify-content: flex-start !important;
        ",
            },
            false,
          ]
        }
      />
    </div>
  </div>
</div>
`;
